pkg/k8scontext/k8scontext_test.go

```diff
@@ -0,0 +1,190 @@
+
+		It("Should be able to follow modifications to the Ingress Resource.", func() {
+			ingress.Spec.Rules[0].Host = "hellow-1.com"
+
+			_, err := k8sClient.Extensions().Ingresses(ingressNS).Update(ingress)
+			Expect(err).Should(BeNil(), "Unabled to update ingress resource due to: %v", err)
+
+			// Retrieve the Ingress to make sure it was updated.
+			ingresses, err := k8sClient.Extensions().Ingresses(ingressNS).List(metav1.ListOptions{})
+			Expect(err).Should(BeNil(), "Unable to retrieve stored ingresses resource due to: %v", err)
+			Expect(len(ingresses.Items)).To(Equal(1), "Expected to have a single ingress stored in mock K8s but found: %d ingresses", len(ingresses.Items))
+
+			// Due to the large sync time we don't expect the cache to be synced, till we force sync the cache.
+			// Start the informers. This will sync the cache with the latest ingress.
+			ctxt.Run()
+
+			ingressListInterface := ctxt.Caches.Ingress.List()
+			// There should still be only one ingress resource.
+			Expect(len(ingressListInterface)).To(Equal(1), "Expected to have a single ingress in the cache but found: %d ingresses", len(ingressListInterface))
+
+			// Retrive the ingresses learnt by the controller.
+			testIngresses := ctxt.GetHTTPIngressList()
+			Expect(len(testIngresses)).To(Equal(1), "Expected to have a single ingress in the k8scontext but found: %d ingresses", len(testIngresses))
+			// Make sure the ingress we got is the ingress we stored.
+			Expect(testIngresses[0]).To(Equal(ingress), "Expected to retrieve the same ingress that we inserted, but it seems we found the following ingress: %v", testIngresses[0])
+
+		})
+
+		It("Should be able to follow modifications to the Ingress Resource.", func() {
+			err := k8sClient.Extensions().Ingresses(ingressNS).Delete(ingressName, nil)
+			Expect(err).Should(BeNil(), "Unable to delete ingress resource due to: %v", err)
+
+			// Retrieve the Ingress to make sure it was updated.
+			ingresses, err := k8sClient.Extensions().Ingresses(ingressNS).List(metav1.ListOptions{})
+			Expect(err).Should(BeNil(), "Unable to retrieve stored ingresses resource due to: %v", err)
+			Expect(len(ingresses.Items)).To(Equal(0), "Expected to have no ingresses stored in mock K8s but found: %d ingresses", len(ingresses.Items))
+
+			// Due to the large sync time we don't expect the cache to be synced, till we force sync the cache.
+			// Start the informers. This will sync the cache with the latest ingress.
+			ctxt.Run()
+
+			ingressListInterface := ctxt.Caches.Ingress.List()
+			// There should still be only one ingress resource.
+			Expect(len(ingressListInterface)).To(Equal(0), "Expected to have no ingress in the cache but found: %d ingresses", len(ingressListInterface))
+
+			// Retrive the ingresses learnt by the controller.
+			testIngresses := ctxt.GetHTTPIngressList()
+			Expect(len(ingressListInterface)).To(Equal(0), "Expected to have no ingress in the k8scontext but found: %d ingresses", len(testIngresses))
+		})
```

Comment:
AFAIK, "It"s can be executed in any order and if this runs earlier, then some of the Modify tests the precedes this can fail. Something to watch out for.

Reply:
The `It` depends on the `BeforeEach` and not any of the other `Its`, the order of the execution of `It` should not matter.